### DISCUSSION TITLE
Embed Vimeo/Youtube

### DISCUSSION THREAD

**Discussion by pixelcrash** [retypeapp/retype#144] [Q&A] [answered]
Hi - how can I embed Vimeo or Youtube Videos? 

I am currently using this code 
`[![Little red ridning hood](http://i.imgur.com/7YTMFQp.png)](https://vimeo.com/3514904 "Little red riding hood - Click to Watch!")`

But this only shows me the image and will direct me to Vimeo. Is there a way of playing the file on the page?

**Comment by fabriciomurta** [comment]
Just copy-paste the "embed" code vimeo makes when you click **Share** in the video, and you should be good.

In your example, it would be:

```html
<iframe src="https://player.vimeo.com/video/3514904?h=b7f24e3b83" width="640" height="360" frameborder="0" allow="autoplay; fullscreen; picture-in-picture" allowfullscreen></iframe>
<p><a href="https://vimeo.com/3514904">Slagsm&aring;lsklubben - Sponsored by destiny</a> from <a href="https://vimeo.com/tomasnilsson">Tomas Nilsson</a> on <a href="https://vimeo.com">Vimeo</a>.</p>
```

**Comment by AndreyChechel** [comment, marked as answer]
Hi @pixelcrash

In addition to the previous answer, there is also a not yet documented `[!embed](link)` component that will help with content embedding. The following attributes are supported (all of them are **optional**):
- `text`: adds a caption for the embedded content
- `el`:  specifies what HTML element should be used, supported values are: `embed`, `iframe`, `video`, `object`. Default is `iframe`
- `height`: sets a custom height value
- `width`:  sets a custom width value
- `aspect`: sets a custom aspect value, supported values are: `1:1`, `4:3`, `16:9`, `21:9`. Default is `16:9`

Retype is smart enough to construct a proper YouTube embed URL based on the provided link. However, Vimeo is not yet supported, so a proper embed URL should be grabbed from their website (click `Share`, and see the iframe `src` value written in the `Embed` section).

```markdown
YouTube video:
[!embed](https://www.youtube.com/embed/qeqhXvE5aPY)

Vimeo video:
[!embed text="Little red riding hood - Click to Watch!"](https://player.vimeo.com/video/3514904?h=b7f24e3b83)
```

There is also the issue  #51 discussing alternative liquid like syntax for the content embedding, which is also supported by Retype.

**Reply by DannyBen** [reply]
One thing that is annoying in the current YouTube embed code (as well as others), is that it is not responsive.

I think that if something is implemented in Retype, it:

a: should *not* be liquid-like, since retype already has its own syntax for special markdown elements.
b: it must be responsive by default.

I usually adapt responsive embed codes from this site:
https://embedresponsively.com/

Of course, it is a little impractical to use the site's output as is in Retype, since it contains a more convoluted HTML, and a CSS snippet - but if this CSS is baked into Retype, then adding some video element should be easy I think.

**Reply by pixelcrash** [reply]
@AndreyChechel thanks it works like a charm with Vimeo and Youtube.

**Reply by AndreyChechel** [reply]
@pixelcrash glad it works for you.

> I usually adapt responsive embed codes from this site:

@DannyBen thanks for sharing that. We'll investigate possibility to make embeds responsive.
